### PR TITLE
Aut 2635/audit events lockout

### PR DESCRIPTION
**Pull request by VladGavrilet**
## What

Add new metadapair for ACCOUNT_TEMPORARILY_LOCKED due to new HMRC requirements

## How to review

Code review

### CODE REVIEW

File: frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/CheckReAuthUserHandler.java [hunk 1]
```diff
@@ -112,7 +112,10 @@ public APIGatewayProxyResponseEvent handleRequestWithUserContext(
                     request.email(),
                     IpAddressHelper.extractIpAddress(input),
                     AuditService.UNKNOWN,
-                    PersistentIdHelper.extractPersistentIdFromHeaders(input.getHeaders()));
+                    PersistentIdHelper.extractPersistentIdFromHeaders(input.getHeaders()),
+                    AuditService.MetadataPair.pair(
+                            "number_of_attempts_user_allowed_to_login",
+                            configurationService.getMaxPasswordRetries()));
```
Comment (BeckaL) [inline frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/CheckReAuthUserHandler.java:118]: I think here which thing it is is going to depend on whether it's a failed number of passwords or a failed number of email attempts - either can throw this error.

File: frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/CheckUserExistsHandler.java [hunk 2]
```diff
@@ -138,7 +138,8 @@ public APIGatewayProxyResponseEvent handleRequestWithUserContext(
                         emailAddress,
                         IpAddressHelper.extractIpAddress(input),
                         AuditService.UNKNOWN,
-                        persistentSessionId);
+                        persistentSessionId,
+                        pair("number_of_attempts_user_allowed_to_login", incorrectPasswordCount));
```
Comment (BeckaL) [inline frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/CheckUserExistsHandler.java:142]: This looks like the wrong thing to me - if we're happy that this should reflect the total number of attempts a user had (as opposed to the number of attempts they've made), then this should be maxPasswordRetries or whatever the thing we check against here is

File: frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/LoginHandler.java [hunk 3]
```diff
@@ -173,7 +173,8 @@ public APIGatewayProxyResponseEvent handleRequestWithUserContext(
                         userProfile.getPhoneNumber(),
                         persistentSessionId,
                         pair("internalSubjectId", userProfile.getSubjectID()),
-                        pair("attemptNoFailedAt", configurationService.getMaxPasswordRetries()));
+                        pair("attemptNoFailedAt", configurationService.getMaxPasswordRetries()),
+                        pair("number_of_attempts_user_allowed_to_login", incorrectPasswordCount));
```
Comment (BeckaL) [inline frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/LoginHandler.java:177]: Same comment here as above

File: frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/LoginHandler.java [hunk 4]
```diff
@@ -216,9 +217,10 @@ public APIGatewayProxyResponseEvent handleRequestWithUserContext(
                             userProfile.getPhoneNumber(),
                             persistentSessionId,
                             pair("internalSubjectId", userProfile.getSubjectID()),
+                            pair("attemptNoFailedAt", configurationService.getMaxPasswordRetries()),
                             pair(
-                                    "attemptNoFailedAt",
-                                    configurationService.getMaxPasswordRetries()));
+                                    "number_of_attempts_user_allowed_to_login",
+                                    configurationService.getCodeMaxRetries()));
```
Comment (BeckaL) [inline frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/LoginHandler.java:223]: To me it looks like this should be `getMaxPasswordRetries`

**Review by BeckaL — COMMENTED** [review]
These look like they need a bit of rethinking. Have we checked with someone that our understanding of what these should reflect (e.g. the max number of attempts configured) is correct?

File: frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/CheckReAuthUserHandler.java [hunk 5]
```diff
@@ -112,7 +116,14 @@ public APIGatewayProxyResponseEvent handleRequestWithUserContext(
                     request.email(),
                     IpAddressHelper.extractIpAddress(input),
                     AuditService.UNKNOWN,
-                    PersistentIdHelper.extractPersistentIdFromHeaders(input.getHeaders()));
+                    PersistentIdHelper.extractPersistentIdFromHeaders(input.getHeaders()),
+                    lockoutType.equals("password")
```
Comment (mark-cab) [inline frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/CheckReAuthUserHandler.java:120]: is the additional variable lockoutType needed? could just use the error code

**Review by mark-cab — APPROVED** [review]

**Pull request merged**